tv_apps/smart-home/js/home.js

```diff
@@ -187,12 +193,96 @@
           cardManager: that.cardManager
         });
         that._cardPicker.on('hide', that.onCardPickerHide.bind(that));
-
-        that.editFolderElem.addEventListener(
-                                        'click', that.onEditFolder.bind(that));
       });
     },
 
+    initContextMenu: function () {
+      this.moveMenuItem.addEventListener('click',
+        () => this.onMoveMenuItemClick());
+
+      this.renameMenuItem.addEventListener('click',
+        () => this.onRenameMenuItemClick());
+
+      this.unpinMenuItem.addEventListener('click',
+        () => this.onUnpinMenuItemClick());
+
+      this.editFolderMenuItem.addEventListener('click',
+        () => this.onEditFolderMenuItemClick());
+    },
+
+    onMoveMenuItemClick: function () {
+      this.edit.toggleEditMode();
+      this.edit.toggleArrangeMode();
+    },
+
+    onRenameMenuItemClick: function () {
+      this.edit.renameCard(this.focusScrollable,
+        this.focusScrollable.getNodeFromItem(this.focusScrollable.currentItem));
+    },
+
+    onUnpinMenuItemClick: function () {
+      this.edit.openDeleteCardDialog(this.focusScrollable,
+        this.focusScrollable.getNodeFromItem(this.focusScrollable.currentItem));
+    },
+
+    onEditFolderMenuItemClick: function () {
+      this._cardPicker.show(this.cardScrollable.currentItem);
+    },
+
+    enableContextMenu: function () {
+      if (this.mode !== '') {
+        return;
+      }
+      var focus = this.spatialNavigator.getFocusedElement();
+      if (focus === this.cardScrollable) {
+        this.cardListElem.setAttribute('contextmenu', this.contextmenuElem.id);
+        this.folderListElem.setAttribute('contextmenu', '');
+        this.cardListElem.style.zIndex = '1';
```

Comment:
Updated to unify the controlling of enabling/disabling context menu through modes, thank you